framework/src/governance/lib/data-catalog-database.ts

```diff
@@ -70,25 +127,64 @@ export class DataCatalogDatabase extends TrackedConstruct {
     const hash = Utils.generateUniqueHash(this);
     this.databaseName = props.name + '_' + hash.toLowerCase();
 
-    let s3LocationUri: string|undefined, locationPrefix: string|undefined;
-
     if (catalogType === CatalogType.S3) {
-      locationPrefix = props.locationPrefix;
 
-      if (!locationPrefix!.endsWith('/')) {
-        locationPrefix += '/';
-      }
+      this.cleanedLocationPrefix = props.locationPrefix === undefined ? undefined : props.locationPrefix.replace(/\/$/g, '');
+      this.s3LocationUri = props.locationBucket!.s3UrlForObject(this.cleanedLocationPrefix);
+
+      if (useLakeFormation) {
+
+        const lfAdmins: IRole[]=[];
+        const cdkRole = Utils.getCdkDeploymentRole(this);
+        lfAdmins.push(cdkRole);
+
+        if (props.permissionModel === PermissionModel.LAKE_FORMATION) {
+          // Create a role for the AwsCustomResource to revoke IAMAllowedPrincipal
+          this.lfRevokeRole = props.lakeFormationConfigurationRole || new Role(this, 'LfRevokeRole', {
+            assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+          });
+          lfAdmins.push(this.lfRevokeRole);
+        }
 
-      s3LocationUri = props.locationBucket!.s3UrlForObject(locationPrefix);
+        this.dataLakeSettings = putDataLakeSettings(this, 'DataLakeSettings', lfAdmins);
+
+        if (props.permissionModel === PermissionModel.LAKE_FORMATION) {
+          this.dataLakeSettings.node.addDependency(this.lfRevokeRole!);
+        }
+
+        // register location
+        if (props.locationBucket) {
+
+          [this.lfDataAccessRole, this.dataLakeLocation] = registerS3Location(
+            this, 'LakeFormationRegistration',
+            props.locationBucket,
+            this.cleanedLocationPrefix,
+            props.permissionModel,
+            props.lakeFormationDataAccessRole,
+          );
+          this.lfDataAccessRole.node.addDependency(this.dataLakeSettings!);
+
+        }
+      }
     }
 
     this.database = new CfnDatabase(this, 'GlueDatabase', {
       catalogId: Stack.of(this).account,
       databaseInput: {
         name: this.databaseName,
-        locationUri: s3LocationUri,
+        locationUri: this.s3LocationUri,
       },
     });
+    this.database.applyRemovalPolicy(removalPolicy);
+
+    if (catalogType === CatalogType.S3 && useLakeFormation) {
+
+      if (props.permissionModel === PermissionModel.LAKE_FORMATION) {
```

Comment:
Why a nested condition? please lets remove the nested condition and check this on line 180 too.